columnphysics/icepack_therm_vertical.F90

```diff
@@ -2457,11 +2455,10 @@ subroutine icepack_step_therm1(dt, ncat, nilyr, nslyr,    &
       smice(:) = c0
       smliq(:) = c0
 
-      allocate(l_meltsliqn(ncat))
-      l_meltsliqn = c0
-      if (present(meltsliqn)) l_meltsliqn = meltsliqn
       l_meltsliq  = c0
+      l_meltsliqn = c0
```

Comment:
OK, this was already initialized below at:
https://github.com/CICE-Consortium/Icepack/blob/67c874194a4acdaf09fc756cd679f579fa53060d/columnphysics/icepack_therm_vertical.F90#L2548-L2555

but I think it makes sense to initialize it here, closer to the beginning of the subroutine. Maybe we'd want to remove that other initialization then ?

Reply:
Actually, this is odd.  In both the prior and current implementation, 

```
l_meltsliqn = c0
if (present(meltsliqn)) l_meltsliqn = meltsliqn
```
then later (line 2555 above)

`l_meltsliqn(n) = c0 
`

so I see no reason to copy meltsliqn into l_meltsliqn if it's ultimately intialized to c0.  The question is whether resetting l_meltsliqn(n) to c0 is a bug AFTER copying in meltsliqn.  Either l_meltsliqn is initialized c0 or meltsliqn.  Right now, it's confusing.  @eclare108213, any thoughts?  I may have messed this up in refactoring at some point, we should clarify and clean up.

Reply:
@apcraig : Looks to me like this first passing of meltsliqn to l_meltsliqn is not needed.  The term l_meltsliqn should be zeroed out once before thermo_vertical. Line 2555 is where it's done in mpas

Reply:
Thanks @njeffery, I'll clean that up.